test/parse.jl

```diff
@@ -41,6 +41,17 @@ Base.iterate(::Issue29451String, i::Integer=1) = i == 1 ? ('0', 2) : nothing
 @test Issue29451String() == "0"
 @test parse(Int, Issue29451String()) == 0
 
+# https://github.com/JuliaStrings/InlineStrings.jl/issues/57
+struct InlineStringIssue57 <: AbstractString end
+Base.ncodeunits(::InlineStringIssue57) = 4
+Base.lastindex(::InlineStringIssue57) = 4
+Base.isvalid(::InlineStringIssue57, i::Integer) = 0 < i < 5
+Base.iterate(::InlineStringIssue57, i::Integer=1) = i == 1 ? ('t', 2) : i == 2 ? ('r', 3) : i == 3 ? ('u', 4) : i == 4 ? ('e', 5) : nothing
+import Base: ==
+==(::SubString{InlineStringIssue57}, x::String) = x == "true"
```

Comment:
may as well overload this one in the same style (also slightly safer not to import `==` i guess)
```suggestion
Base.:(==)(::SubString{InlineStringIssue57}, x::String) = x == "true"
```